modules/apps/document-library/document-library-item-selector-web/src/main/java/com/liferay/document/library/item/selector/web/internal/display/context/DLItemSelectorViewDisplayContext.java

```diff
@@ -364,8 +368,39 @@ private Long _getFileEntryTypeId() {
 		InfoItemItemSelectorCriterion infoItemItemSelectorCriterion =
 			(InfoItemItemSelectorCriterion)_itemSelectorCriterion;
 
-		return GetterUtil.getLong(
-			infoItemItemSelectorCriterion.getItemSubtype());
+		long fileEntryTypeId = GetterUtil.getLong(
+			infoItemItemSelectorCriterion.getItemSubtype(), -1);
+
+		DLFileEntryType dlFileEntryType = null;
+
+		if (fileEntryTypeId >= 0) {
+			dlFileEntryType = _dlFileEntryTypeLocalService.fetchFileEntryType(
+				fileEntryTypeId);
+		}
+
+		if (dlFileEntryType == null) {
+			fileEntryTypeId = _getFileEntryTypeId(
+				infoItemItemSelectorCriterion.getItemSubtype());
+		}
+
+		return fileEntryTypeId;
+	}
+
+	private long _getFileEntryTypeId(String itemSubtype) {
```

Comment:
I would rename the input parameter to `fileEntryTypeKey` to better understand what is expected in the method

Reply:
Done